src/scripts/main.js

```diff
@@ -1,3 +1,33 @@
 'use strict';
 
-// write code here
+const list = Array.from(document.querySelectorAll('ul li'));
+
+function sortList() {
+  function parseSalary(salaryString) {
+    return parseFloat(salaryString.replace(/[$,]/g, ''));
+  }
+
+  list.sort((a, b) => {
+    const salaryA = parseSalary(a.dataset.salary);
+    const salaryB = parseSalary(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+
+  const ul = document.querySelector('ul');
+
+  list.forEach(li => ul.appendChild(li));
+}
+
+function getEmployees() {
+  return list.map(li => ({
+    name: li.textContent,
+    position: li.dataset.position,
+    salary: li.dataset.salary,
+    age: parseInt(li.dataset.age),
+  }));
+}
+
+sortList(list);
```

Comment:
sortList function does not require any arguments. Remove the argument 'list' when calling the function.